packages/lib-classifier/src/store/WorkflowStepStore.js

```diff
@@ -0,0 +1,87 @@
+import { autorun } from 'mobx'
+import { addDisposer, getRoot, types } from 'mobx-state-tree'
+import Step from './Step'
+import { SingleChoiceTask, MultipleChoiceTask } from './Task'
+
+const WorkflowStepStore = types
+  .model('WorkflowStepStore', {
+    active: types.maybe(types.reference(Step)),
+    steps: types.maybe(types.map(Step)),
+    tasks: types.maybe(types.map(types.union((snapshot) => {
+      if (snapshot.type === 'SingleChoiceTask') return SingleChoiceTask;
+      if (snapshot.type === 'MultipleChoiceTask') return MultipleChoiceTask;
+    }, SingleChoiceTask, MultipleChoiceTask)))
+  })
+  .views(
+    get activeStepTasks() {
+      if (self.active) {
+        return self.active.taskKeys.map((taskKey) => {
+          self.tasks.get(taskKey)
+        })
+      }
+
+      return []
+    }
+  )
+  .actions(self => {
+    function afterAttach() {
+      createWorkflowObserver()
+    }
+
+    function reset() {
+      self.active = null
+      self.steps = null
+      self.tasks = null
+    }
+
+    function setStepsAndTasks(workflow) {
+      const taskKeys = Object.keys(workflow.tasks)
+      self.steps = workflow.steps
+      if (taskKeys.length > 0) {
+        taskKeys.forEach((taskKey) => {
+          // Set tasks object as a MobX observable JS map in the store
+          self.tasks.set(taskKey, workflow.tasks[taskKey])
```

Comment:
I might want to use `put` here instead of `set`, but the only place I see this described in in the MST docs about Map types: https://github.com/mobxjs/mobx-state-tree/blob/master/API.md#typesmap

I don't see it in the MobX docs itself...

Reply:
Looking over the source for MobX and MST, it appears to only be for MST.